lib/Zonemaster/Backend/DB.pm

```diff
@@ -130,6 +133,30 @@ sub process_unfinished_tests {
     }
 }
 
+sub generate_deterministic_hash {
+    my ( $self, $params ) = @_;
+
+    my $profile = Zonemaster::Engine::Profile->effective;
+
+    my %to_encode = ();
+    $to_encode{domain}      = lc $$params{domain}   // "";
+    $to_encode{ds_info}     = $$params{ds_info}     // [];
+    $to_encode{ipv4}        = $$params{ipv4}        // $profile->get( 'net.ipv4' );
+    $to_encode{ipv6}        = $$params{ipv6}        // $profile->get( 'net.ipv6' );
+    $to_encode{nameservers} = $$params{nameservers} // [];
+    $to_encode{priority}    = $$params{priority}    // 10;
```

Comment:
This already gets a default value in RCP.pm. Needed here? -- The same applies for "queue" below.

Reply:
I choose to define a default value for all entries because I tried to be cautious. I need to make some tests and look deeper into the code to check which values are already set or not when reaching this part of the code.

Reply:
see https://github.com/zonemaster/zonemaster-backend/pull/798#discussion_r655516858